app.py

```diff
@@ -1,55 +1,72 @@
-from flask import Flask, request
-from flask_restful import reqparse, abort, Api, Resource
-from flask_pymongo import PyMongo
+from flask import Flask, jsonify
+from flask_restful import Resource, Api, reqparse, request, abort
+from model import Model
 
 app = Flask(__name__)
-mongo = PyMongo(app)
 api = Api(app)
 
-
-USERS = {
-    'user1': {'name': 'elena'},
-    'user2': {'name': 'bob'},
-}
-
-def abort_if_user_doesnt_exist(user_id):
-    if user_id not in USERS:
-        abort(404, message="User {} doesn't exist".format(user_id))
-
 parser = reqparse.RequestParser()
 parser.add_argument('name')
 
-class Users(Resource):
-    def get(self, user_id):
-        abort_if_user_doesnt_exist(user_id)
-        return USERS[user_id]
+class Student(Resource):
+    def get(self, student_id):
+        try:
+            student=Model.get_by_id(student_Id)
+        except:
+            abort(404, message="Getting Student {} failed".format(student_id))
+        return student
 
-    def delete(self, user_id):
-        abort_if_user_doesnt_exist(user_id)
-        del USERS[user_id]
+    def delete(self, student_id):
+        try:
+            deleted = Model.delete(student_Id)
+        except:
+            abort(404, message="Deleting Student {} failed".format(student_id))
         return '', 204
+    
+    def post(self):
+        args = parser.parse_args()
+        try:
+            n= {'name': args['name']}
```

Comment:
If `name` is not within args you will get a KeyError Exception thrown .. check the parameter you require exists when doing your payload validations